packages/clerk-js/src/ui/components/ApiKeys/ApiKeysTable.tsx

```diff
@@ -0,0 +1,189 @@
+import { useClerk } from '@clerk/shared/react';
+import type { ApiKeyResource } from '@clerk/types';
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+
+import { timeAgo } from '../../../utils/date';
+import {
+  Button,
+  descriptors,
+  Flex,
+  Icon,
+  Input,
+  localizationKeys,
+  Spinner,
+  Table,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+} from '../../customizables';
+import { ThreeDotsMenu } from '../../elements';
+import { useClipboard } from '../../hooks';
+import { Check, Copy, Eye, EyeSlash } from '../../icons';
+
+const useApiKeySecret = ({ apiKeyID, enabled }: { apiKeyID: string; enabled: boolean }) => {
+  const clerk = useClerk();
+
+  return useSWR(enabled ? ['api-key-secret', apiKeyID] : null, () => clerk.getApiKeySecret(apiKeyID));
+};
+
+const CopySecretButton = ({ apiKeyID }: { apiKeyID: string }) => {
+  const [enabled, setEnabled] = useState(false);
+  const { data: apiKeySecret } = useApiKeySecret({ apiKeyID, enabled });
+  const { onCopy, hasCopied } = useClipboard(apiKeySecret ?? '');
+
+  useEffect(() => {
```

Comment:
why are we doing this in an effect?

Reply:
ah i see we only allow copying when the input is revealed. Is that for security reasons? I'd assume I should still be able to copy even when the input is masked.

Reply:
Hey! Yeah by default the endpoint does not include the secret key for security reasons 👀 We're fetching it when the eye-icon is clicked or the copy button is clicked.

Reply:
ah interesting, I need to pull and play around with it. What does the copy button do before your reveal it, is it disabled?

Reply:
Let me know if you're gonna play with it so we can add you to the instance 👀 

> What does the copy button do before your reveal it, is it disabled?

Same as just clicking the reveal icon button, fetch then copy to clipboard. Not disabled.